renderlib/SceneView.h

```diff
@@ -0,0 +1,62 @@
+#pragma once
+
+#include "CCamera.h"
+
+#include <memory>
+
+class Scene;
+class RenderSettings;
+
+// collect up a Scene, CCamera, and Viewport region for rendering
+struct SceneView
+{
+  // viewport region is min value at lower left, max value at upper right of window
+  struct Viewport
+  {
+    struct Region
+    {
+      Region()
+        : lower(+INT_MAX)
+        , upper(-INT_MAX)
+      {
+      }
+
+      Region(const glm::ivec2& lower, const glm::ivec2& upper)
+        : lower(lower)
+        , upper(upper){};
+
+      // assignment operator
+      Region& operator=(const Region& other)
+      {
+        lower = other.lower;
+        upper = other.upper;
+        return *this;
+      }
+
+      void extend(const glm::ivec2& p)
+      {
+        lower = glm::min(lower, p);
+        upper = glm::max(upper, p);
+      }
+
+      glm::ivec2 size() const { return upper - lower; }
+
+      bool empty() const { return size().x < 0 || size().y < 0; }
```

Comment:
Nit: isEmpty vs. empty?

Reply:
Also should this be <= comparison?

Reply:
precedent for using the name `empty()` is in C++ STL containers.   But I get it, it's a controversial choice.  I guess empty is in the sense of an adjective and not a verb.